utils/src/main/scala/za/co/absa/enceladus/utils/implicits/DataFrameImplicits.scala

```diff
@@ -68,6 +69,29 @@ object DataFrameImplicits {
       SparkUtils.withColumnIfDoesNotExist(df, colName, col)
     }
 
+    /**
+     * Set nullable property of column.
+     *
+     * @param columnName is the column name to change
+     * @param nullable   boolean flag to set the nullability of the column `columnName` to
+     */
+    def withNullableColumnState(columnName: String, nullable: Boolean): DataFrame = {
+      // Courtesy of https://stackoverflow.com/a/33195510
+
+      // modify [[StructField] with name `columnName` if its nullability differs
+      val newSchema = StructType(df.schema.map {
+        case StructField(c, t, n, m) if c.equals(columnName) && n != nullable =>
+          StructField(c, t, nullable = nullable, m)
+        case y: StructField => y
+      })
+
+      if (newSchema == df.schema) {
+        df // no change
+      } else {
+        df.sqlContext.createDataFrame(df.rdd, newSchema) // apply new schema
```

Comment:
A very paranoid thought: Could it never happen that the `errCol` is `null` , but the settings says it should not be nullable?

Reply:
I don't think that it is currently possible.

Reply:
OK, cool. I trust you 😉